backend/business-partner-agent/src/main/java/org/hyperledger/bpa/model/BPAPresentationExchange.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2020-2021 - for information on the respective copyright owner
+ * see the NOTICE file and/or the repository at
+ * https://github.com/hyperledger-labs/business-partner-agent
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hyperledger.bpa.model;
+
+import io.micronaut.data.annotation.AutoPopulated;
+import io.micronaut.data.annotation.DateCreated;
+import io.micronaut.data.annotation.TypeDef;
+import io.micronaut.data.model.DataType;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.hyperledger.aries.api.issue_credential_v1.CredentialExchangeRole;
+import org.hyperledger.aries.api.issue_credential_v1.CredentialExchangeState;
+import org.hyperledger.aries.api.present_proof.PresentProofRequest;
+import org.hyperledger.aries.api.present_proof.PresentationExchangeRecord;
+import org.hyperledger.aries.api.present_proof.PresentationExchangeRole;
+import org.hyperledger.aries.api.present_proof.PresentationExchangeState;
+import org.hyperledger.bpa.api.CredentialType;
+
+import javax.annotation.Nullable;
+import javax.persistence.*;
+import java.time.Instant;
+import java.util.Map;
+import java.util.UUID;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Entity
+@Table(name = "bpa_presentation_exchange")
+public class BPAPresentationExchange {
+    
+    @Id
+    @AutoPopulated
+    private UUID id;
+    
+    @DateCreated
+    private Instant createdAt;
+        
+    @OneToOne
+    private Partner partner;
+    
+    @Nullable
+    private String label;
+    
+    private String threadId;
+    
+    private String presentationExchangeId;
+    
+    @Enumerated(EnumType.STRING)
+    private PresentationExchangeRole role;
+    
+    @Enumerated(EnumType.STRING)
+    private PresentationExchangeState state;
+    
+    private Map<String, Object> presentation;
+
+    private Map<String, Object> presentationProposalDict;
+
+    @TypeDef(type = DataType.JSON)
+    private PresentProofRequest.ProofRequest presentationRequest;
```

Comment:
Running into an issue with this setup. 

On saving, it looks great. and the db value looks like this. 

```
{"name": "Proof request", "nonce": "877537186890811073056529", "version": "1.0", "requestedAttributes": {"referent": {"names": ["attr1", "attr2"], "restrictions": [{"schema_id": "95Ro4QRFU8AygJ9ZgfaXGE:2:test_schema:0.1"}]}}, "requestedPredicates": {}}
```
but when loading it from the database requestedAttributes[0]['referent']['restrictions'] is empty.

@etschelp  how should I be saving/loading this attribute?

Reply:
I see, thats basically the same issue as above. That's my bad as I annotated the models only with the serialize part and missed the deserialize part. I will fix this in the wrapper, and after a version upgrade that should be fixed.

Reply:
try client version 0.20.0

Reply:
that fixed it.thanks.